### PR TITLE
WEB-12610: Update for MySQL 8

### PR DESCRIPTION
**Pull request by davidsickmiller**
# Key change

Have `make mysql` produce a file containing MySQL 8 files instead of MySQL 5.7 files.

# How this was done

Merged in the latest changes from megastep:master.

# Also in this PR

We previously deleted the code related to other database systems like postgresql.  This complicated the merge, and is likely to complicated future merges, so I've restored the deleted files (and merged in their latest changes).

Locked Dockerfile.mysql to pull specifically from version 2 of amazonlinux (see https://github.com/megastep/mysql-lambda/issues/6)

Configured `zip` to preserve symlinks, to save disk space once expanded.

Updated our readme.

# To test

Run `make mysql` and confirm you get a zip file.  Examine the contents of the zip file and make sure they seem reasonable.

Put that zip file in asknicelydo, deploy to staging, and confirm things that use mysql, mysqldump, zip, and unzip are working.  (I believe this can be done later after approving this PR.)

### CODE REVIEW

**Comment by davidsickmiller** [comment]
Looks like this builds for ARM architecture; let's see if I can change it to AMD...

**Comment by askfilmon** [comment]
I get these error running `make mysql`

Looks like I need a login
```
Unable to find image 'mysql-layer:latest' locally
Error response from daemon: pull access denied for mysql-layer, repository does not exist or may require 'docker login': denied: requested access to the resource is denied
```

```
docker container cp :/root/mysql-8.0.32-layer.zip .
must specify at least one container source
make: *** [mysql-8.0.32-layer.zip] Error 1
```

**Comment by davidsickmiller** [comment]
> I get these error running `make mysql`
> 
> Looks like I need a login
> ```
> Unable to find image 'mysql-layer:latest' locally
> Error response from daemon: pull access denied for mysql-layer, repository does not exist or may require 'docker login': denied: requested access to the resource is denied
> ```
> 
> ```
> docker container cp :/root/mysql-8.0.32-layer.zip .
> must specify at least one container source
> make: *** [mysql-8.0.32-layer.zip] Error 1
> ```

Try running:
docker container create mysql-layer

**Comment by askfilmon** [comment]
> > I get these error running `make mysql`
> > Looks like I need a login
> > ```
> > Unable to find image 'mysql-layer:latest' locally
> > Error response from daemon: pull access denied for mysql-layer, repository does not exist or may require 'docker login': denied: requested access to the resource is denied
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > ```
> > docker container cp :/root/mysql-8.0.32-layer.zip .
> > must specify at least one container source
> > make: *** [mysql-8.0.32-layer.zip] Error 1
> > ```
> 
> Try running: docker container create mysql-layer

Thanks, that worked

File: README.md [hunk 1]
```diff
@@ -1,41 +1,62 @@
...
+and these errors:
+```
+docker container cp :/root/mysql-8.0.32-layer.zip .                                                                                                                                                                                  
+must specify at least one container source                                                                                                                                                                                           
+make: *** [mysql-8.0.32-layer.zip] Error 1
+```
+
+Try running this before the `make mysql`:
+```
+docker container create mysql-layer
```
Comment (askfilmon) [inline README.md:58]: 😆

**Review by askfilmon — APPROVED** [review]
👍🏾

**Pull request merged**